lib/stoplight/data_store/memory.rb

```diff
@@ -1,62 +1,53 @@
 # coding: utf-8
 
-require 'concurrent'
+require 'monitor'
 
 module Stoplight
   module DataStore
     # @see Base
     class Memory < Base
+      include MonitorMixin
+
       def initialize
-        @failures = Concurrent::Map.new { [] }
-        @states = Concurrent::Map.new { State::UNLOCKED }
-        @lock = Monitor.new
+        @failures = Hash.new { |h, k| h[k] = [] }
+        @states = Hash.new { |h, k| h[k] = State::UNLOCKED }
+        super() # MonitorMixin
       end
 
       def names
-        (all_failures.keys + all_states.keys).uniq
+        synchronize { @failures.keys | @states.keys }
       end
 
       def get_all(light)
-        [get_failures(light), get_state(light)]
+        synchronize { [get_failures(light), get_state(light)] }
```

Comment:
Although it appears to be fine to next synchronize calls, it'd be simpler to not use the get_* methods here